hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java

```diff
@@ -555,6 +556,10 @@ SqlResult execute(SelectPlan plan, QueryId queryId, List<Object> arguments, long
                     }
 
                     partitionKeyComponents[i] = perMapCandidate.get(attribute).eval(null, evalContext);
+                    // handling case of __key = ? where the parameter is an PartitionAware OBJECT.
+                    if (partitionKeyComponents[i] instanceof PartitionAware) {
```

Comment:
@Fly-Style  don't forget to support that in https://github.com/hazelcast/hazelcast/pull/25006/files#diff-86ab7c5e4dd6cacb2de0651945c4cbb729ac6d46ccd5398adcfe120b91563986R83